core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -795,6 +795,14 @@ public boolean supportBooleanCaseWhen() {
     return true;
   }
 
+  /**
+   * Returns whether to enable unparse of "macro-like" calls such as IS TRUE.
+   * "A IS TRUE" can be unparse to "A IS NOT NULL AND A" When A is deterministic.
+   * Otherwise, inconsistent results may occur. */
+  public boolean supportMacroLikeUnparse() {
```

Comment:
I don't understand how this fixes the problem of non-deterministic functions.

Reply:
This has not been fixed. It would be more accurate to say that it is a kind of compatible approach, which refers to SqlValidator#withCallRewrite.

Reply:
I think this is not really necessary then.

Reply:
Ok. This approach is somewhat inappropriate.